### DISCUSSION TITLE
StreamOpen before connected possible?

### DISCUSSION THREAD

**Discussion by timprepscius** [microsoft/msquic#1604] [Q&A] [answered]
Can I call StreamOpen before receiving a callback for:

    case QUIC_CONNECTION_EVENT_CONNECTED:

Or do I have to wait until after the connection has connected.

Thanks,

-tim

**Comment by nibanks** [comment, marked as answer]
Good timing. I just created this PR today: https://github.com/microsoft/msquic/pull/1600. It should answer your question, but if you have any clarifying questions/comments I'd appreciate you comment on that PR.

P.S. The answer is **yes**, you can open, start and even send on a stream before the connection is connected (or even started).

**Reply by timprepscius** [reply]
Great- thank you very much.
I have one more question, will start another discussion thread.
